File: cache_interceptor/src/main/java/uk/co/glass_software/android/cache_interceptor/interceptors/error/ErrorCode.java
package uk.co.glass_software.android.cache_interceptor.interceptors.error;

public enum ErrorCode {
    
    NETWORK(true),
    UNAUTHORISED(false),
    NOT_FOUND(false),
    UNEXPECTED_RESPONSE(true),
    UNKNOWN(true);
    
    public final boolean isRecoverable;
    
    ErrorCode(boolean isRecoverable) {
        this.isRecoverable = isRecoverable;
    }
}

File: cache_interceptor/src/main/java/uk/co/glass_software/android/cache_interceptor/interceptors/error/ErrorCode.kt
package uk.co.glass_software.android.cache_interceptor.interceptors.error

enum class ErrorCode constructor(val canRetry: Boolean) {
    NETWORK(true),
    UNAUTHORISED(false),
    NOT_FOUND(false),
    UNEXPECTED_RESPONSE(true),
    SERVER_ERROR(true),
    UNKNOWN(true)
}
